yjit/src/codegen.rs

```diff
@@ -215,6 +219,13 @@ impl<'a> JITState<'a> {
     // Check if we are compiling the instruction at the stub PC
     // Meaning we are compiling the instruction that is next to execute
     pub fn at_current_insn(&self) -> bool {
+        // If this is not the first block compiled by gen_block_series(),
+        // it might be compiling the same block again with a different Context.
+        // In that case, it should defer_compilation() and inspect the stack there.
+        if !self.first_block {
```

Comment:
This is a bit weird, because this is not about the PC anymore. Maybe we should improve the name of this method. Not a big deal though.

Reply:
Also kind of curious to have an example of how the same block would get compiled again with a different context? Does this mean twice the same block in the same sequence? Like a loop getting unrolled?

Reply:
>  Maybe we should improve the name of this method.

Sure, I'll come up with something after having a lunch break.

> Also kind of curious to have an example of how the same block would get compiled again with a different context? Does this mean twice the same block in the same sequence?

I included an example in `test_yjit`, which originally lived in rubyspec. The `Foo#==` calls itself with a different `reg_mapping` at `opt_eq`.

Reply:
> Maybe we should improve the name of this method.

Given that it's essentially a guard for `defer_compilation`, how about `jit.at_compile_target()` https://github.com/ruby/ruby/pull/11280/commits/acd8cd72de740503ddfb7ec634212ab19456b728?